### PR TITLE
Huge update to modernize the code

### PR DESCRIPTION
**Pull request by aperloff**
This PR covers several tasks:
1. It updates all the python scripts to be python 3 compatible. Most, if not all, of them will not be python 2 backwards compatible.
2. It updates the formatting of the python files to conform to the pylint style (`.pylintrc` file now contained in the repo).
3. It updates the formatting of the shell scripts to conform to best practices as laid out by [ShellCheck](https://github.com/koalaman/shellcheck/wiki).

The code that has been checked for functionality so far:
- [x] GetPythonVersions.py
- [x] GetSiteInfo.py
- [x] RecursiveFileList.py
- [x] containerize/containerize.sh
- [x] copy_certificates
- [x] copyfiles.py
- [x] eosdu
- [x] install_certificate
- [x] install_certificate_for_cernsso
- [x] make_certificate_file
- [x] toolgenie.py
- [x] xrdcp_recursive.py

I will need the help from others to make sure all of these changes are good.

@mtonjes @gbenelli @kpedro88 @jhakala @PerilousApricot

### CODE REVIEW

**Comment by aperloff** [comment]
Thanks @kpedro88. I appreciate you taking a look. I think I'll turn this into a unit test, though the path will be need to be different so that I can provide some reference values. Perhaps `/store/user/cmsdas/test/` is a good folder to use as it will probably be static over a long period of time.

**Comment by aperloff** [comment]
@kpedro88 I have added to the README some instructions on how to semi-manually unit-test `eosdu` and other bash scripts.

https://github.com/FNALLPC/lpc-scripts/blob/ff24c91ccc8be8c68359849a605711b63a0f7b82/README.md

**Comment by aperloff** [comment]
I've added the capability to run pytests on some of the python modules. These tests are now stored in the tests directory. Information on how to install and run these tests has also been added to the README.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,82 @@
...
+
+*Please note, these tests do not constitute complete coverage. Some additional manual testing may be necessary.*
+
+#### Bats for Bash scripts
+
+The [Bats](https://bats-core.readthedocs.io/en/stable/) tests are currently setup to test only the `eosdu` executable. Every effort has been made to test all of the options. Even so, full coverage is not guaranteed. These tests rely on the eos path `/store/user/cmsdas/test/` being stable.
+
+```bash
+cd <path to lpc-scripts>/lpc-scripts
+git clone https://github.com/bats-core/bats-core.git test/bats
```
Comment (kpedro88) [inline README.md:20]: Proposal:
1. create a `tests` directory in this repo
2. put these clone commands in a script `setup_bash.sh`
3. put the test running command in a script `run_bash.sh`
4. similar approach for the manual python tests below

Comment (aperloff) [reply README.md]: Done. Slightly different naming than you suggested and a unified setup/run script, but the same idea in principle. Let me know what you think.Done. Slightly different naming than you suggested and a unified setup/run script, but the same idea in principle. Let me know what you think.

File: test/pytest_control.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    -s      Setup the virtual environment
+EOF
+}
+
+OPTIONS=""
+REMOVE="False"
+SETUP="False"
+
+#check arguments
+while getopts "hors" option; do
```
Comment (kpedro88) [inline test/pytest_control.sh:43]: should it be `o:`?

File: test/pytest_control.sh [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+setup() {
+    python3 -m venv test/venv
+    # shellcheck disable=SC1091
+    source test/venv/bin/activate
+    pip install pytest six
+}
+
+teardown() {
+    if [[ "$VIRTUAL_ENV" != "" ]]; then
+        die "You must call 'deactivate' before removing the virtual environment"
```
Comment (kpedro88) [inline test/pytest_control.sh:17]: can't you just make this function call deactivate if the venv is active?

File: test/pytest_control.sh [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    esac
+done
+
+if [[ "${SETUP}" == "True" ]]; then
+    setup_pytest_venv
+elif [[ "${REMOVE}" == "True" ]]; then
+    teardown_pytest_venv
+else
+    if [[ "$VIRTUAL_ENV" == "" ]]; then
+        error_pytest_control "The Python virtual environment is not setup"
```
Comment (kpedro88) [inline test/pytest_control.sh:77]: why not just do `source test/venv/bin/activate` inside the script in this case? then the user doesn't have to source it or take other manual actions

File: RecursiveFileList.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+"""Print a map of the directory/file structure of a given base path.
+The module will print out a full map in dictionary format, a list of the files with their full paths,
+and a list of directories with their full paths. The module can be run using the following command:
+
+python3 RecursiveFileList.py <absolute path>
+
+It can also be imported by another python program and used like:
+
+#import RecursiveFileList
+RecursiveFileList.map_dir("<absolute path>")
```
Comment (jhakala) [inline RecursiveFileList.py:12]: `map_dir` takes two arguments, an enclosing folder and a folder within it -- so instead I would suggest two examples like:
``` RecursiveFileList.map_dir("<absolute path to enclosing folder> / <subfolder>")```
and
``` RecursiveFileList.map_dir("", "<absolute path to folder>")```

File: test/pytest_control.sh [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        error_pytest_control "The directory '${PWD}/test/venv' does not exist. Cannot remove the virtual environment"
+    fi
+}
+
+usage() {
+cat <<EOF
+usage: pytest_control.sh [options]
+
+This script sets up or tears down (removes) the software needed to run the pytest unit/integration tests
+on the python modules in FNALLPC/lpc-scrips.
```
Comment (kpedro88) [inline test/pytest_control.sh:53]: typo: scrips -> scripts

File: test/pytest_control.sh [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+        # shellcheck disable=SC1091
+        source test/venv/bin/activate
+        pip install pytest six
+    else
+        error_pytest_control "The directory '${PWD}/test/venv' does not exist. Cannot enter the virtual environment"
+    fi
+}
+
+run_pytest() {
+    if [[ -f "test/venv/bin/activate" ]]; then
```
Comment (kpedro88) [inline test/pytest_control.sh:21]: should you also check if the virtualenv is already active?

File: test/pytest_control.sh [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+        source test/venv/bin/activate
+    fi
+
+    if [[ "$VIRTUAL_ENV" != "" ]]; then
+        if [[ -f "test/test.py" ]]; then
+            IFS=' '; read -r -a OPTIONS_ARRAY <<<"${1}"
+            pytest test/test.py "${OPTIONS_ARRAY[@]}"
+        else
+            error_pytest_control "Unable to locate the file containing the tests to run ('test/test.py')." \
+                                 "Make sure the virtual environment was setup from within the lpc-scrips directory"
```
Comment (kpedro88) [inline test/pytest_control.sh:32]: typo: scrips -> scripts

File: test/pytest_control.sh (hunk 6 above)
Comment (aperloff) [reply test/pytest_control.sh]: Fixed

File: test/pytest_control.sh (hunk 8 above)
Comment (aperloff) [reply test/pytest_control.sh]: Fixed

File: test/pytest_control.sh (hunk 7 above)
Comment (aperloff) [reply test/pytest_control.sh]: I added the check, but it shouldn't really matter. By executing the script it's performed in a subprocess, which won't have the virtualenv active even if it is active on the parent process.

**Pull request merged**